libs/docs/src/docs/2.2.0/create-devfiles-with-templates.md

```diff
@@ -0,0 +1,186 @@
+---
+title: Creating Devfiles With Templates
+description: Creating Devfiles With Templates
+---
+
+Most dev tools can utilize the devfile registry to 
+fetch the stacks needed to begin development on projects. 
+Sometimes, however, you might need to create a devfile 
+from scratch or to create your own devfile template(s). 
+This guide runs through the process, starting from a minimum devfile and 
+building sample templates for common use cases.
+
+## Prerequisites
+
+- [Versions](./versions)
+- [Metadata](./metadata)
+- [Adding components](./adding-components)
+- [Adding commands](./adding-commands)
+
+## Procedure
+
+1. Creating a minimal devfile
+    - `schemaVersion` is the only required root element
+    - `metadata` is optional but it is recommended to have in your 
+    templates
+    ```yaml {% title="Minimal Devfile" filename="devfile.yaml" %}
+    schemaVersion: 2.2.0
+    ```
+    ```yaml {% title="Minimal Devfile with Metadata" filename="devfile.yaml" %}
+    schemaVersion: 2.2.0
+    metadata:
+      name: devfile-sample
+      version: 2.0.0
+    ```
+2. Creating a web service template
+    1. Template `metadata`
+        - Set the template `name` and `version`
+        - Set a `description` for the template
+        - Set the `projectType` and `language` to describe what
+        kind of stack is being used
+        - Set `provider` to tell who is providing this template
+        - Set `tags` to give keyword which describe the elements
+        of the stack
+        - Set `architectures` to specify the platforms support in 
+        the template
+        - For improved readability on devfile registries, set `displayName` to the title that will be the display text for this template and `icon` to tie a stack icon to the template:
+        ```yaml
+          schemaVersion: 2.2.0
+          metadata:
+            name: web-service
+            version: 1.0.0
+            description: A web service template.
+            projectType: Go
+            language: Go
+            provider: Red Hat
+            tags: [ 'Go', 'Gin', 'pq' ]
+            architectures: [ 'amd64' ]
+            displayName: Simple Web Service
+            icon: https://raw.githubusercontent.com/devfile-samples/devfile-stack-icons/main/golang.svg
+          ```
+    2. Setup `components`
+        - A `name` is required for a component.
```

Comment:
I would go with a single style for list items, when they are complete sentences - the first part of the list does not use periods at the end of each item, while the second one does.

Reply:
@gabriel-rh I changed these instances to use sub-lists rather than multiple sentences and removed ending periods on all list items.